### PR TITLE
Determine_molecules: Modularization + {Partition+Split+Merge}

### PR DESCRIPTION
**Pull request by afshinfard**
Added splitting and merging so we can perform clustering on sub-sub-graphs.
All stages of determine_molecule is now Modularized so they return a list of set, each set as a community.
and some other minor changes.

### CODE REVIEW

**Comment by afshinfard** [comment]
@sjackman I tested the changes, and I confirm that the results are promising. The new strategies added are extremely faster, and can produce the result of the same quality (with good parameters).
I also noticed that pycharm or whatever has changed some lines of the code for their formatting (like applying standard spacing within commands). The changes are well made, but they're for parts that other's contributed. Is that okay?

**Comment by afshinfard** [comment]
@sjackman and also some of the changes are only a reformatting and modularizing of the previous code. So now all the functions related to barcode to molecule separation return in the same format, and we can simply use them beside each other to make/test new pipelines (as I did for strategies 1 to 5 and 10).

**Comment by sjackman** [comment]
> The changes are well made, but they're for parts that other's contributed. Is that okay?

Please make those changes in a separate PR. Once that PR is accepted into `master`, you can rebase this PR onto `master`.

File: physlr/physlr.py [hunk 1]
```diff
@@ -37,6 +38,7 @@ def progress_bar_for_file(fin):
         mininterval=1, smoothing=0.1,
         bar_format="{percentage:4.1f}% {elapsed} ETA {remaining} {bar}")
 
+
```
Comment (sjackman) [inline physlr/physlr.py]: Only one blank line between functions please.

File: physlr/physlr.py [hunk 2]
```diff
@@ -1021,107 +1024,190 @@ def assign_read_molecule(mxs, moltomxs, mol_counts, bx):
...
 
     @staticmethod
     def determine_molecules(g, u, strategy):
-        "Assign the neighbours of this vertex to molecules."
-        if strategy == 2:
-            return Physlr.determine_molecules_k_clique_communities(g, u)
-        if strategy == 3:
-            return Physlr.determine_molecules_louvain(g, u)
-        if strategy == 4:
-            return Physlr.determine_molecules_cosine_of_squared(g, u)
```
Comment (sjackman) [inline physlr/physlr.py]: Please add these functions `Physlr.determine_molecules_*`. They can be defined in terms of `Physlr.community_detection_*` if you like. For example:
```python
def determine_molecules_k_clique_communities(g, u):
    return [community
        for bi_connected_component in
        Physlr.community_detection_biconnected_components(g, set(g.neighbors(u)))
        for community in
        Physlr.community_detection_k_clique(g, bi_connected_component)]
end
```

**Comment by sjackman** [comment]
> some of the changes I made are only a reformatting and modularization of the previous code.

Please open a separate PR to refactor existing code before adding new code. After that PR is merged into `master`, this PR can be rebased onto `master`.

File: physlr/physlr.py (hunk 1 above)
Comment (afshinfard) [reply physlr/physlr.py]: Oh okay, pycharm reports warning for one blank line and asks for 2. I'll change it.

**Comment by sjackman** [comment]
Has this PR been superseded by PR #62?

**Comment by sjackman** [comment]
If so click the `Delete branch` button above.

**Pull request closed without merging**